File: smartdevices-app/smartdevicesapp.model/src/main/java/de/vogler_engineering/smartdevicesapp/model/entities/dto/DeviceConfig.java
/**
 * Copyright (c) Vogler Engineering GmbH. All rights reserved.
 * Licensed under the MIT License. See LICENSE.md in the project root for license information.
 */
package de.vogler_engineering.smartdevicesapp.model.entities.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

import de.vogler_engineering.smartdevicesapp.model.entities.tabs.TabConfig;
import lombok.Data;

/**
 * Created by vh on 14.02.2018.
 */

@Data
public class DeviceConfig {
    private AppInfo info;

    private String deviceId;
    private String deviceName;
    private String user;
    private String dataUrl;

    private boolean reconfigure;

    @JsonProperty("tabs")
    private List<TabConfig> tabConfig;
}

File: smartdevices-gateway/SmartDevicesGateway.Model/Internal/DeviceConfig.cs
//
// Copyright (c) Vogler Engineering GmbH. All rights reserved.
// Licensed under the MIT License. See LICENSE.md in the project root for license information.
//
using System;
using System.Collections.Generic;
using System.Text;
using SmartDevicesGateway.Model.Config.SDConfig;
using SmartDevicesGateway.Model.Dto.Config;
using SmartDevicesGateway.Model.Ui;

namespace SmartDevicesGateway.Model.Internal
{
    public class DeviceConfig
    {
        public AppInfo Info { get; set; }
        public int UpdateInterval { get; set; }
        public DeviceId DeviceId { get; set; }
        public User User { get; set; }

        public IEnumerable<DeviceGroup> DeviceGroups { get; set; }

        public ISet<string> Values { get; set; }

        public ISet<string> Actions { get; set; }

        public ISet<string> Tabs { get; set; }
        public SmartDevice Device { get; set; }
    }
}
